### PR TITLE
Allow Exporting Specification and Vocabulary as RDF.

### PR DESCRIPTION
**Pull request by yashkhatri**
Dependent on OSLC4j-Bundle in lyo.core.

### CODE REVIEW

**Comment by jadelkhoury** [comment]
Is this the complete patch to have export functionality working?
Any special instructions to build? Any other patches I need to build first?

**Comment by berezovskyi** [comment]
@yashkhatri which PR shall I merge first?

File: org.eclipse.lyo.tools.emf2rdf/src/main/java/org/eclipse/lyo/tools/emf2rdf/constants/ExportConstants.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @since 2.4.0-SNAPSHOT
+ */
+public class ExportConstants {
+
+       public static final String OSLC_SHAPE = "Oslc Shape";
+       public static final String SHACL_SHAPE = "Shacl Shape";
+       public static final String EXPORT_DESTINATION = "exportDestination";
+       public static final String FILE = "File";
+       public static final String TRIPLESTORE = "Triple Store";
+       public static final String SPARQL_UPDATE_END_POINT = "sparqlUpdateEndpPoint";
```
Comment (berezovskyi) [inline org.eclipse.lyo.tools.emf2rdf/src/main/java/org/eclipse/lyo/tools/emf2rdf/constants/ExportConstants.java]: I dont think the conn string shall be part of the export...

File: org.eclipse.lyo.tools.emf2rdf/src/main/java/org/eclipse/lyo/tools/emf2rdf/handler/ConvertHandler.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+public class ConvertHandler extends ActionDelegate implements IActionDelegate {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConvertHandler.class);
+    protected List<IFile> files;
+
+ /*
+ * This methods are used if we use menu contribution instead of object contribution in plugin.xml.
+ * And replace "extends AbstractHandler implements IHandler" in this class. 
+ * 
+    protected String getPersistentProperty(IResource res, QualifiedName qn) {
```
Comment (berezovskyi) [inline org.eclipse.lyo.tools.emf2rdf/src/main/java/org/eclipse/lyo/tools/emf2rdf/handler/ConvertHandler.java]: should not have dead code here

**Comment by jadelkhoury** [comment]
@yashkhatri any updates on this feature?

**Comment by yashkhatri** [comment]
[osgi-bundle] (https://github.com/eclipse/lyo.core/pull/16) is merged. This PR requires a code review and verification and then we can merge.

**Comment by berezovskyi** [comment]
Jad, are you taking care of this review? Or shall I review some parts too (only if you need help)?

**Comment by jadelkhoury** [comment]
This review is on me. It has taken some time to be able to deploy but I hope we solved all problems now.

**Comment by jadelkhoury** [comment]
A incremental list of feedback and actions to deal with:

- [ ] Add the EMF model from which some of the java classes are generated (Good for future changes)
- [ ] the pom.xml file of the Emf2rdf project refers to ${version.lyo}. No other pom file in the repository does so. Replace with the explicit 2.4.0-SNAPSHOT. (@berezovskyi any reason we have hardcoded the version number everywhere?)
- [ ] Currently, the dialogs prompt for 2 seperate files for Vocab and Specs. Instead, prompt for a folder, and not file names. Give exported files default names such as OslcDomainSpecification.ttl ShaclDomainSpecification.ttl and vocabulary.ttl. This becomes 1 instead of 3 prompts.
- [ ] When user selects export menu item, they are first given a wizard that ends with "Finish". But then the prompt for file location happens much later. Instead, prompt hte user with ALL information before triggering the code for export. 
- [ ] An export that results in errors, simply gives the message "export unsuccesful". Instead, given more information, or point user to window (log? error? console?) for more information.

**Comment by berezovskyi** [comment]
@jadelkhoury seems like you are right about the versions!

@jadelkhoury @jamsden our releases can become much easier with https://stackoverflow.com/questions/19123013/maven-version-with-a-property

**Comment by yashkhatri** [comment]
@jadelkhoury @berezovskyi Regarding third point:

_"Currently, the dialogs prompt for 2 seperate files for Vocab and Specs. Instead, prompt for a folder, and not file names. Give exported files default names such as OslcDomainSpecification.ttl ShaclDomainSpecification.ttl and vocabulary.ttl. This becomes 1 instead of 3 prompts."_

When prompting for asking the file name, there are three possibilities in which format the file can be exported i.e. turtle (.ttl), N-Triples (.nt), and RDF/XML (.rdf)

screen shot 2018-09-10 at 22 41 30

If we remove this prompt and export files with default names, we will loose this capability to export in different formats.

What do you think?

**Comment by jadelkhoury** [comment]
I did not notice this option actually. 
I wouldn’t tie the rdf format to the file extension.  I would rather have a specific dialog that asks for format, but that would mean an additional dialog/logic. 

For now, I can even live with a single format (turtle) just to get this feature out. We can add the option later! But if easy to prompt, offer it now.

**Pull request closed without merging**